algolia/algoliasearch-client-javascript
# chore(lint): add linting

**Summary**

This PR adds formatting and linting to the project:

- Prettier and ESLint (with the [Algolia configuration](https://github.com/algolia/eslint-config-algolia))
- EditorConfig

Linting can be performed manually. Additionally, every attempt to commit triggers linting (with Husky and lint-staged).

The idea is to ensure we don't commit anything "dirty" and avoid breaking a build because of linting errors. This reduces (if not completely removes) the need for "style" commits.

**Result**

To test this, try to:

- Add a JavaScript or TypeScript file with incorrect formatting or rule violations.
- Run `yarn lint`, this should raise errors.
- Stage the faulty file and try to commit it, this should raise errors as well.
- Run `yarn lint:fix`, this should take care of fixable issues.

![lint](https://user-images.githubusercontent.com/5370675/60455176-10785780-9c36-11e9-89fa-bc4e10e28db8.gif)
